Tests/SentryProfilerTests/SentryProfilerSwiftTests.swift

```diff
@@ -207,121 +203,92 @@ class SentryProfilerSwiftTests: XCTestCase {
 
     func testConcurrentProfilingTransactions() throws {
         let numberOfTransactions = 10
         var spans = [Span]()
 
-        for _ in 0 ..< numberOfTransactions {
-            spans.append(try fixture.newTransaction())
-        }
+        func createConcurrentSpansWithMetrics() throws {
+            for _ in 0 ..< numberOfTransactions {
+                let span = try fixture.newTransaction()
+                spans.append(span)
+                fixture.currentDateProvider.advanceBy(nanoseconds: 100)
+            }
 
-        forceProfilerSample()
+            forceProfilerSample()
 
-        for (i, span) in spans.enumerated() {
-            try fixture.gatherMockedMetrics(span: span)
-            span.finish()
+            for (i, span) in spans.enumerated() {
+                try fixture.gatherMockedMetrics(span: span)
+                span.finish()
 
-            try self.assertValidProfileData()
-            try self.assertMetricsPayload(metricsBatches: i + 1)
+                try self.assertValidProfileData()
+                try self.assertMetricsPayload(metricsBatches: i + 1)
+            }
         }
 
+        try createConcurrentSpansWithMetrics()
+
         // do everything again to make sure that stopping and starting the profiler over again works
         spans.removeAll()
-        fixture.currentDateProvider.reset()
 #if !os(macOS)
         fixture.resetGPUExpectations()
         fixture.framesTracker.resetFrames()
-#endif
-        fixture.prepareMetricsMocks()
-
-        for _ in 0 ..< numberOfTransactions {
-            spans.append(try fixture.newTransaction())
-        }
-
-        forceProfilerSample()
-
-#if !os(macOS)
         fixture.displayLinkWrapper.call()
 #endif
-        for (i, span) in spans.enumerated() {
-            try fixture.gatherMockedMetrics(span: span)
-            span.finish()
 
-            try self.assertValidProfileData()
-            try self.assertMetricsPayload(metricsBatches: i + 1)
-        }
+        try createConcurrentSpansWithMetrics()
     }
 
     /// Test a situation where a long-running span starts the profiler, which winds up timing out, and then another span starts that begins a new profile, then finishes, and then the long-running span finishes; both profiles should be separately captured in envelopes.
     /// ```
-    ///    time                0s                         1s     2s     2.5s     3s  (these times are adjusted to the 1s profile timeout for testing only)
+    ///                        time->
     ///    transaction A       |---------------------------------------------------|
     ///    profiler A          |---------------------------x  <- timeout
     ///    transaction B                                           |-------|
     ///    profiler B                                              |-------|  <- normal finish
     ///   ```
     func testConcurrentSpansWithTimeout() throws {
-        let originalTimeoutInterval = kSentryProfilerTimeoutInterval
```

Comment:
The rest of the test cases are modified to use the mocked clock and timeout timer instead of relying on DispatchQueue and test expectations.